docs/sagas.md

```diff
@@ -0,0 +1,239 @@
+---
+id: sagas
+title: Sagas
+---
+
+A saga describes a long running process as a sequence of events.
+
+## Sagas Overview
+
+You can define a saga as a set of event handler functions. Each such function runs in response to a specific event and can do the following:
+
+- Send a command to an aggregate
+- Schedule a command on the specified moment in time
+- Store intermediate data in a persistent storage
+- Trigger a side effect
+
+This allows you to organize branching chains of events and side effects to describe processes of any complexity. For example, the code below demonstrates a saga that organizes a web site's user registration process:
+
+<!-- prettier-ignore-start -->
+
+[mdis]:# (../tests/saga-sample/saga.js)
+```js
+export default {
+  handlers: {
+    Init: async ({ store }) => {
+      await store.defineTable('users', {
+        indexes: { id: 'string' },
+        fields: ['mail']
+      })
+    },
+    USER_CREATED: async ({ store, executeCommand }, event) => {
+      await store.insert('users', {
+        id: event.aggregateId,
+        mail: event.payload.mail
+      })
+      await executeCommand({
+        aggregateName: 'User',
+        aggregateId: event.aggregateId,
+        type: 'requestConfirmUser',
+        payload: event.payload
+      })
+    },
+    USER_CONFIRM_REQUESTED: async ({ sideEffects, scheduleCommand }, event) => {
+      await sideEffects.sendEmail(event.payload.mail, 'Confirm mail')
+      await scheduleCommand(event.timestamp + 1000 * 60 * 60 * 24 * 7, {
+        aggregateName: 'User',
+        aggregateId: event.aggregateId,
+        type: 'forgetUser',
+        payload: {}
+      })
+    },
+    USER_FORGOTTEN: async ({ store }, event) => {
+      await store.delete('users', {
+        id: event.aggregateId
+      })
+    }
+  },
+  sideEffects: {
+    sendEmail: async (mail, content) => {
+      ...
+    }
+  }
+}
+```
+
+<!-- prettier-ignore-end -->
+
+This saga requires a new user to confirm their email address. If a user does not confirm the address in a week, the saga cancels the registration.
+
+## Define a Saga
+
+### Add a Saga to the Application
+
+You can define a saga in one of the following two ways:
```

Comment:
Change to "You can define a saga in the following ways:"

Reply:
done